api/v1beta2/types.go

```diff
@@ -356,12 +356,53 @@ type IBMCloudResourceReference struct {
 	Name *string `json:"name,omitempty"`
 }
 
+// IBMCloudCatalogOffering represents an IBM Cloud Catalog Offering resource.
+// +kubebuilder:validation:XValidation:rule="(has(self.offeringCRN) && !has(self.versionCRN)) || (!has(self.offeringCRN) && has(self.versionCRN))",message="either offeringCRN or version CRN must be provided, not both"
+type IBMCloudCatalogOffering struct {
+	// OfferingCRN defines the IBM Cloud Catalog Offering CRN. Using the OfferingCRN expects that the latest version of the Offering will be used.
+	// If a specific version should be used instead, rely on VersionCRN.
+	// +optional
+	OfferingCRN *string `json:"offeringCRN,omitempty"`
+
+	// PlanCRN defines the IBM Cloud Catalog Offering Plan CRN to use for the Offering.
+	// +optional
+	PlanCRN *string `json:"planCRN,omitempty"`
+
+	// VersionCRN defines the IBM Cloud Catalog Offering Version CRN. A specific version of the Catalog Offering will be used, as defined by this CRN.
+	// +optional
+	VersionCRN *string `json:"versionCRN,omitempty"`
+}
+
 // NetworkInterface holds the network interface information like subnet id.
 type NetworkInterface struct {
+	// SecurityGroups defines a set of IBM Cloud VPC Security Groups to attach to the network interface.
+	// +optional
+	// +k8s:conversion-gen=false
+	SecurityGroups []VPCResource `json:"securityGroups,omitempty"`
+
 	// Subnet ID of the network interface.
 	Subnet string `json:"subnet,omitempty"`
 }
 
+// VPCLoadBalancerBackendPoolMember represents a VPC Load Balancer Backend Pool Member.
+type VPCLoadBalancerBackendPoolMember struct {
+	// LoadBalancer defines the Load Balancer the Pool Member is for.
+	// +required
+	LoadBalancer VPCResource `json:"loadBalancer"`
+
+	// Pool defines the Load Balancer Pool the Pool Member should be in.
+	// +required
+	Pool VPCResource `json:"pool"`
+
+	// Port defines the Port the Load Balancer Pool Member listens for traffic.
+	// +required
+	Port int64 `json:"port"`
```

Comment:
Any reasons for defining Port as int64 and wieght as pointer?

Reply:
Yes, Port is required for LB member creation, for which we do not expect to always use `6443`.
And weight is not required, hence it is made optional.